### ISSUE TITLE
DetailsList: Accessibility - Narrator doesn't read checkmark column header in single-select mode in Edge 

### ISSUE THREAD

**Issue by eveleens** [OfficeDev/office-ui-fabric-react#6576]
### Bug Report 
 
 
- __Package version(s)__: (fill this out) at least 6.4
- __Browser and OS versions__: (fill this out if relevant) Edge
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No) No
 
Requested priority: (Blocking, High, Normal, Low) High
 
Products/sites affected: (if applicable) Flow 
 
#### Describe the issue:
 
When using Scan mode in Narrator, the Narrator doesn't read the 'select all' checkbox or any column header that has no title. 
So in compact - https://developer.microsoft.com/en-us/fabric#/components/detailslist/compact , it'll skip to "Name", instead of stopping at the 'select all' header, which is technically the first column. It'll read "Name" as the second column correctly though. 
In Grouped custom - https://developer.microsoft.com/en-us/fabric#/components/detailslist/customitemcolumns , it'll skip to "key" (skipping the 'select all' checkbox and the icon column). 

Funnily, it does the correct thing in Chrome. 
 
#### Actual behavior:
 
It doesn't read the columns with no text visible. 

#### Expected behavior:
 
It should read something for all columns.

**Comment by eveleens** [comment]
Should be fixed, according to #6646

**Comment by eveleens** [comment]
This bug still exists when you're in the "single" select mode in the DetailsList. A user can't take any action there, but it should still read the column header (checkboxes).

**Comment by betrue-final-final** [comment]
The check box could be disabled, which makes it non-navigable, or hidden, which is also non-navigable.

**Comment by natalieethell** [comment]
Spoke to Ben offline and it seems like making the "select all" checkbox disabled in single selection mode is the way to go. It shouldn't be tab navigable, but you should be able to arrow to it via table navigation in scan mode. 

With a fix, the screen reader ([for the example here](https://developer.microsoft.com/en-us/fabric#/components/detailslist/variablerowheights)) should read out something like the following if a user arrows to the "select all" checkbox in scan mode in single selection mode:

**Column 1, Toggle selection for all items, disabled**

**Comment by natalieethell** [comment]
Okay, update! So it seems like the main issue you brought up (not being able to focus on the "select all" checkbox in multi-select mode) is fixed. 

As for the "select all" checkbox acting as the column header for checkboxes in single selection mode:

I just confirmed that the "select all" checkbox is actually already using the disabled styling for DetailsList checks, which actually prevents the checkbox from being focused. I spoke to Ben about this more offline and it sounds like we'd like to keep it this way.

That does mean that in single selection mode, the checkboxes still have column index 1, but no focusable column header. @cschlechty what do you think, is this alright? We probably still want the checkboxes to have a column index, right?

**Comment by natalieethell** [comment]
Okay so! I spoke with @cschlechty offline and it sounds like we want the checkboxes to have a column index. In single selection mode, the "select all" checkbox should not exist. Instead, we can render something simple as the column header for those checkboxes. I should be able to get to this next week, thank you for your patience!
